### PR TITLE
EXERCISE 2 FIRST ATTEMPT

### PR DESCRIPTION
**Pull request by josh231101**
== FEATURES ==
* Sidebar (Sidebar is not yet finished but it's included)
* Navbar
* Hero section with animation using grid
* Main content
* 1- Grid beautiful cards 
* 2.- How it works section with 4 cards
* 3.- Mobile experience with iphone mockup
* 4.- Coupon section with a video slightly overlapped
* Footer 
Images:
image
image
image
image
image
image

### CODE REVIEW

File: Josue/index.html [hunk 1]
```diff
@@ -12,20 +12,402 @@
...
+        <div class="coupon-section__wrapper">
+          <h1 class="wt">Get Free Coupon Now</h1>
+          <p class="wt">
+            Lorem ipsum dolor, sit amet consectetur adipisicing elit. Eos
+            asperiores eligendi, esse eius sit odit.Lorem ipsum dolor, sit amet
+            consectetur adipisicing elit. Eos asperiores eligendi, esse eius sit
+            odit
+          </p>
+          <form class="coupon-section__form" action="POST">
+            <input
```
[the remark this answers is not in the document]
Comment (josh231101) [reply Josue/index.html:376]: I was following the design, UI designer forgot accessibility design

File: Josue/css/styles.css [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+strong {
+  color: black;
+}
+footer {
+  padding: 40px 80px;
+}
+/*UTILS*/
+.btn {
+  width: fit-content;
+  padding: 16px 52px;
```
Comment (frank-martinez-iest) [inline Josue/css/styles.css]: preferably keep px's to be a factor of 8 (48px or 56px) so we can have consistency

File: Josue/css/styles.css [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+  border: none;
+  border-radius: 54px;
+  background-color: #57a772;
+  color: white;
+  font-weight: 500;
+  font-size: 12px;
+  cursor: pointer;
+}
+.btn:hover {
+  transition: all 400ms ease-out;
```
Comment (frank-martinez-iest) [inline Josue/css/styles.css:61]: 💯

File: Josue/css/styles.css [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+  animation: pop-in 0.6s cubic-bezier(0, 0.9, 0.3, 1.2) forwards;
+  width: 100%;
+  height: 100%;
+  background: #57a772;
+  color: white;
+}
+.sidebar--closed {
+  display: none;
+}
+.sidebar--open {
```
Comment (frank-martinez-iest) [inline Josue/css/styles.css:103]: You mastered BEM!! 🤟

File: Josue/css/styles.css [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+.navbar__logo {
+  flex: 1;
+}
+.navbar__items {
+  display: flex;
+  align-items: center;
+  width: 100%;
+}
+.navbar__link {
+  margin: 0 8px;
```
Comment (frank-martinez-iest) [inline Josue/css/styles.css]: NIT: If you set the horizontal margin to 16px looks better, don't you think? 
image

File: Josue/css/styles.css [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+  border-radius: 8px;
+}
+.card__buttons > span,
+i {
+  margin-left: 8px;
+}
+.icon--first {
+  flex: 1;
+}
+/* HOW IT WORKS SECTION*/
```
Comment (frank-martinez-iest) [inline Josue/css/styles.css:243]: I loved the comments, very useful

File: Josue/css/styles.css [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+  grid-template-columns: 1fr 1fr;
+}
+.download-section__left {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin-bottom: 32px;
+}
+/*COUPON SECTION*/
+.coupon-section {
```
Comment (frank-martinez-iest) [inline Josue/css/styles.css]: NIT: I would prefer this class to be called coupon or coupon-container, not sure I've seen the word `section` being used as it seems redundant or vague.

File: Josue/js/app.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+const sidebar = document.querySelector(".sidebar");
+
+const openSideBar = () => {
```
Comment (frank-martinez-iest) [inline Josue/js/app.js:3]: This is a good JS exercise, but there is a CSS only way to achieve this, have you seen it?

File: Josue/index.html [hunk 9]
```diff
@@ -12,21 +12,456 @@
...
   </head>
 
   <body>
-    <!-- Add your site or application content here -->
-    <p>Hello world! This is HTML5 Boilerplate.</p>
-    <script src="js/vendor/modernizr-{{MODERNIZR_VERSION}}.min.js"></script>
+    <!--SIDEBAR-->
+    <aside class="sidebar sidebar--closed">
+      <i
+        arialabel=""
```
Comment (frank-martinez-iest) [inline Josue/index.html]: I think this is not needed

File: Josue/index.html [hunk 10]
```diff
@@ -12,21 +12,456 @@
...
+            <p class="info-card__description">
+              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ducimus
+              quis culpa facere inventore harum esse illum
+            </p>
+          </li>
+        </ul>
+      </section>
+
+      <!-- MOBILE EXPERIENCE SECTION -->
+      <section class="section-container download-section">
```
Comment (frank-martinez-iest) [inline Josue/index.html]: lets try to come up with a better name for this element, it is a section but the classname is `section-container` which is redundant

File: Josue/index.html [hunk 11]
```diff
@@ -12,21 +12,456 @@
...
+            ipsam obcaecati. Labore dolore maxime sequi debitis cumque maiores
+            vero ullam?
+          </p>
+          <button class="btn">Download on the App Store</button>
+        </div>
+        <img src="./images/iphone.png" alt="iphone mockup with mifood app" />
+      </section>
+
+      <!-- COUPON SECTION -->
+      <section class="section-container coupon-section">
```
Comment (frank-martinez-iest) [inline Josue/index.html]: same as above

File: Josue/index.html [hunk 12]
```diff
@@ -12,21 +12,456 @@
...
+        <div class="coupon-section__wrapper">
+          <h1 class="wh">Get Free Coupon Now</h1>
+          <p class="wh">
```
Comment (frank-martinez-iest) [inline Josue/index.html]: same as above, wh is not clear what it means

File: Josue/index.html [hunk 13]
```diff
@@ -12,21 +12,456 @@
...
+            <li><a href="#">Press</a></li>
+            <li><a href="#">Blog</a></li>
+            <li><a href="#">Careers</a></li>
+            <li><a href="#">Security</a></li>
+            <li><a href="#">Driver Centre</a></li>
+          </ul>
+        </li>
+        <li>
+          <p class="footer-title">Information</p>
+          <hr class="hr--green" />
```
Comment (frank-martinez-iest) [inline Josue/index.html:432]: we probably could have gotten away with css to not need an HR to see a line here but lets keep it like this

**Review by frank-martinez-iest — CHANGES_REQUESTED** [review]
So much progress in such short time! Im proud of you, great job! Just left a couple comments here and there

File: Josue/css/styles.css (hunk 5 above)
Comment (josh231101) [reply Josue/css/styles.css]: Yup even better

File: Josue/css/styles.css (hunk 7 above)
Comment (josh231101) [reply Josue/css/styles.css]: I saw it here: https://zova.com/
But indeed it's redundant, gonna change it

File: Josue/index.html (hunk 13 above)
Comment (josh231101) [reply Josue/index.html:432]: border-bottom, I thought of it but the green line according to the design wasn't taking the width of the text

File: Josue/index.html (hunk 10 above)
Comment (josh231101) [reply Josue/index.html]: Doubt: Should I change the name of section-container or download-section? Or both :laughing: ??
The section-container is css code is:
`.section-container {
  box-sizing: border-box;
  width: 100%;
  padding: 80px;
  margin-left: auto;
  margin-right: auto;
}`
Should I call the section something like -> container-wrapper??

File: Josue/js/app.js (hunk 8 above)
Comment (josh231101) [reply Josue/js/app.js:3]: I saw some codepen projects, many of them with sass or less, and the ones with pure css use the burger icon as an input type checkbox so when is active they add some CSS, should I do that or leave it with JS?

File: Josue/index.html [hunk 14]
```diff
@@ -12,21 +12,456 @@
...
+            />
+            <button class="btn btn--black" type="submit">Subscribe Now</button>
+          </form>
+        </div>
+      </section>
+
+      <!-- VIDEO SECTION -->
+      <section class="section-container video-section">
+        <video
+          class="video-section__content"
```
Comment (josh231101) [inline Josue/index.html]: container__video-content

**Pull request merged**